lib/ansible/modules/cloud/amazon/ec2_vpc_net_facts.py

```diff
@@ -57,77 +60,183 @@
+            description: True/False depending on attribute setting for DNS support.
+            returned: always
+            type: boolean
+        ipv6_cidr_block_association_set:
+            description: An array of IPv6 cidr block association set information.
+            returned: always
+            type: complex
+            contains:
+                association_id:
+                    description: The association ID
+                    returned: always
+                    type: string
+                ipv6_cidr_block:
+                    description: The IPv6 CIDR block that is associated with the VPC.
+                    returned: always
+                    type: string
+                ipv6_cidr_block_state:
+                    description: A hash/dict that contains a single item. The state of the cidr block association.
+                    returned: always
+                    type: dict
+                    contains:
+                        state:
+                            description: The CIDR block association state.
+                            returned: always
+                            type: string
+'''
 
+import traceback
 from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.ec2 import connect_to_aws, ec2_argument_spec, get_aws_connection_info
+from ansible.module_utils.ec2 import (
+    boto3_conn,
+    ec2_argument_spec,
+    get_aws_connection_info,
+    AWSRetry,
+    HAS_BOTO3,
+    boto3_tag_list_to_ansible_dict,
+    camel_dict_to_snake_dict,
+    ansible_dict_to_boto3_filter_list
+)
 
+try:
+    from botocore.exceptions import ClientError, NoCredentialsError
+except ImportError:
+    pass  # caught by imported HAS_BOTO3
 
-def get_vpc_info(vpc):
 
-    try:
-        classic_link = vpc.classic_link_enabled
-    except AttributeError:
-        classic_link = False
+@AWSRetry.backoff()
+def describe_vpcs(connection, module):
+    """
+    Describe VPCs
 
-    vpc_info = { 'id': vpc.id,
-                 'instance_tenancy': vpc.instance_tenancy,
-                 'classic_link_enabled': classic_link,
-                 'dhcp_options_id': vpc.dhcp_options_id,
-                 'state': vpc.state,
-                 'is_default': vpc.is_default,
-                 'cidr_block': vpc.cidr_block,
-                 'tags': vpc.tags
-               }
+    connection  : boto3 client connection object
+    module  : AnsibleModule object
+    """
 
-    return vpc_info
+    # collect parameters
+    filters = ansible_dict_to_boto3_filter_list(module.params.get('filters'))
+    vpc_ids = module.params.get('vpc_ids')
 
-def list_ec2_vpcs(connection, module):
+    # init empty list for return vars
+    vpc_info = list()
+    vpc_list = list()
 
-    filters = module.params.get("filters")
-    vpc_dict_array = []
+    # Get the basic VPC info
+    try:
+        response = connection.describe_vpcs(VpcIds=vpc_ids, Filters=filters)
+    except (ClientError, NoCredentialsError) as e:
```

Comment:
I don't believe so. I previously had this as just catching ClientError but updated per a comment on another PR and to bring it inline with other boto3 facts modules. I cruelly don't think we need to catch NoCredentialsError at all since it is ultimately a ClientError.